src/rules/require-gitignore.ts

```diff
@@ -0,0 +1,12 @@
+import { buildRule } from './build-rule';
+import { RuleName } from './types';
+import { fileConforms } from '../rule-helpers';
+
+export default buildRule({
+  name: RuleName.RequireGitIgnore,
+  description: 'Is `.gitignore` present, and does it conform?',
+  dependencies: [],
+  execute: async (ruleExecutionArguments) => {
+    return await fileConforms('.gitignore', ruleExecutionArguments);
```

Comment:
In the ticket, I had suggested that we perform a partial match by checking that the projects's version of `.gitignore` starts with the template's version. This would allow projects to add more lines to `.gitignore` without violating lint. However, I am not 100% sure that this would be necessary yet. Most of the projects which are already in alignment do not have extra lines in `.gitignore`, and the ones which aren't have a completely different `.gitignore`. I think it would be valuable to only check that Yarn v3 lines are present if Yarn v3 is installed, but that seems like a separate ticket. All that to say... I think this change is good for now and we can fine-tune in a future PR, when we review how well these rules work for real-world projects.